source/cellular_pktio.c

```diff
@@ -470,23 +472,34 @@ static _atRespType_t _getMsgType( CellularContext_t * pContext,
         {
             if( ( pContext->PktioAtCmdType != CELLULAR_AT_NO_COMMAND ) && ( inputWithSrcPrefix == true ) )
             {
+                /* Celluar interface is sending AT command and this line contains
+                 * expected prefix in the response. Return AT_SOLICITED here. */
                 atRespType = AT_SOLICITED;
             }
             else
             {
+                /* Lines with prefix are considered AT_UNSOLICITED unless the prefix
+                 * is expected in AT command response. */
                 atRespType = AT_UNSOLICITED;
             }
         }
         else
         {
-            if( ( ( pContext->PktioAtCmdType != CELLULAR_AT_NO_COMMAND ) && ( pRespPrefix == NULL ) ) ||
-                ( pContext->PktioAtCmdType == CELLULAR_AT_MULTI_DATA_WO_PREFIX ) ||
-                ( pContext->PktioAtCmdType == CELLULAR_AT_WITH_PREFIX ) ||
-                ( pContext->PktioAtCmdType == CELLULAR_AT_MULTI_WITH_PREFIX ) ||
-                ( pContext->PktioAtCmdType == CELLULAR_AT_WITH_PREFIX_NO_RESULT_CODE ) )
```

Comment:
Why do you replace these conditions with `CELLULAR_AT_NO_COMMAND` only?

Reply:
The logic here is to handle token without prefix. The original implementation covers all the AT command type here.
```
typedef enum CellularATCommandType
{
    CELLULAR_AT_NO_RESULT,                  /**<  no response expected, only OK, ERROR etc. */
    CELLULAR_AT_WO_PREFIX,                  /**<  string response without a prefix. */
    CELLULAR_AT_WITH_PREFIX,                /**<  string response with a prefix. */
    CELLULAR_AT_MULTI_WITH_PREFIX,          /**<  multiple line response all start with a prefix. */
    CELLULAR_AT_MULTI_WO_PREFIX,            /**<  multiple line response with or without a prefix. */
    CELLULAR_AT_MULTI_DATA_WO_PREFIX,       /**<  multiple line data response with or without a prefix. */
    CELLULAR_AT_WO_PREFIX_NO_RESULT_CODE,   /**<  string response without a prefix and no result code is expected. */
    CELLULAR_AT_WITH_PREFIX_NO_RESULT_CODE, /**<  string response WITH a prefix and no result code is expected. */
    CELLULAR_AT_NO_COMMAND                  /**<  no command is waiting response. */
} CellularATCommandType_t;
```


```
( pContext->PktioAtCmdType != CELLULAR_AT_NO_COMMAND ) && ( pRespPrefix == NULL )
```
=>
```
    CELLULAR_AT_NO_RESULT,                  /**<  no response expected, only OK, ERROR etc. */
    CELLULAR_AT_WO_PREFIX,                  /**<  string response without a prefix. */
    CELLULAR_AT_WITH_PREFIX,                /**<  string response with a prefix. */
    CELLULAR_AT_WO_PREFIX_NO_RESULT_CODE,   /**<  string response without a prefix and no result code is expected. 
```

```
( pContext->PktioAtCmdType == CELLULAR_AT_MULTI_DATA_WO_PREFIX ) ||
                ( pContext->PktioAtCmdType == CELLULAR_AT_WITH_PREFIX ) ||
                ( pContext->PktioAtCmdType == CELLULAR_AT_MULTI_WITH_PREFIX ) ||
                ( pContext->PktioAtCmdType == CELLULAR_AT_WITH_PREFIX_NO_RESULT_CODE ) )
```
=>
```
    CELLULAR_AT_MULTI_WITH_PREFIX,          /**<  multiple line response all start with a prefix. */
    CELLULAR_AT_MULTI_WO_PREFIX,            /**<  multiple line response with or without a prefix. */
    CELLULAR_AT_MULTI_DATA_WO_PREFIX,       /**<  multiple line data response with or without a prefix. */
    CELLULAR_AT_WITH_PREFIX_NO_RESULT_CODE, /**<  string response with a prefix and no result code is expected. */
```

These conditions can be simplified with `( pContext->PktioAtCmdType != CELLULAR_AT_NO_COMMAND )`. The `pRespPrefix` will be checked in `_Cellular_PktioSendAtCmd`.